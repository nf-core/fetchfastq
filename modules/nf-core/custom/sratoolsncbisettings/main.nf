process CUSTOM_SRATOOLSNCBISETTINGS {
    tag 'ncbi-settings'
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/sra-tools:3.0.8--h9f5acd7_0' :
        'biocontainers/sra-tools:3.0.8--h9f5acd7_0' }"

    input:
    List ids

    output:
    Path ncbi_settings = path('*.mkfg')

    topic:
    [ task.process, 'sratools', eval("vdb-config --version 2>&1 | grep -Eo '[0-9.]+'") ] >> 'versions'

    shell:
    config = "/LIBS/GUID = \"${UUID.randomUUID().toString()}\"\\n/libs/cloud/report_instance_identity = \"true\"\\n"
    template 'detect_ncbi_settings.sh'
}
