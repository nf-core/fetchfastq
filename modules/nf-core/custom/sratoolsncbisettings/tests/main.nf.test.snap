{
    "Should run without failures": {
        "content": [
            [
                "versions.yml:md5,3d6ee88cce1ee517e198633f062589a8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:17:35.745689"
    }
}